### PR TITLE
Add partial update option for typespec java emitter

### PR DESCRIPTION
**Pull request by thang-bit**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/riskiq/Easm/tspconfig.yaml [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
   "service-dir":
     default: "sdk/easm"
+  "python-sdk-folder":
+    default: "{project-root}/azure-sdk-for-python/"
+  "java-sdk-folder": 
+    default: "{project-root}/azure-sdk-for-java/"
+  "js-sdk-folder":
+    default: "{project-root}/azure-sdk-for-js/"
+  "csharp-sdk-folder":
+    default: "{project-root}/azure-sdk-for-net/"
```
Comment (weidongxu-microsoft) [inline specification/riskiq/Easm/tspconfig.yaml:11]: I don't think we need these xx-folder?
sample config https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml

File: specification/riskiq/Easm/tspconfig.yaml [hunk 2]
```diff
@@ -17,11 +25,15 @@ options:
...
+    emitter-output-dir: "{csharp-sdk-folder}/sdk/{service-dir}/{package-dir}"
   "@azure-tools/typespec-ts":
     package-dir: "defendereasm-rest"
     packageDetails:
       name: "@azure-rest/defender-easm-rest"
+    emitter-output-dir: "{js-sdk-folder}/sdk/{service-dir}/{package-dir}"
   "@azure-tools/typespec-java":
     package-dir: "azure-analytics-defender-easm"
     namespace: com.azure.analytics.defender.easm
+    emitter-output-folder: "{java-sdk-folder}/sdk/{service-dir}/{package-dir}"
```
Comment (weidongxu-microsoft) [inline specification/riskiq/Easm/tspconfig.yaml:37]: again, I assume we do not need this `emitter-output-folder`?

**Review by weidongxu-microsoft — APPROVED** [review]
I've approved it. You can use this commit to test Java first. If it works and everything works fine, ping me to merge.

**Comment by thang-bit** [comment]
@weidongxu-microsoft, do I need to wait for this PR to merge into the main branch first? Or I can just go ahead and regenerate the sdk?

**Comment by weidongxu-microsoft** [comment]
You can go ahead using the last commit of this (latest be 219fcd35385531e2af9c5b210ea195b8d05b393b?) to re-generate SDK.

After everything good, we can merge it and you made one last update to tsp-location.yaml in SDK.

**Review by haolingdong-msft — APPROVED** [review]
LGTM, you can try to regenerate sdk using this tspconfig.yaml to make sure everything works fine.

**Pull request merged**